tools/wake-format/emitter.cpp

```diff
@@ -201,23 +201,60 @@ static size_t count_trailing_newlines(const token_traits_map_t& traits, const CS
   return it->second.after_bound.size();
 }
 
-static size_t count_allowed_newlines(const token_traits_map_t& traits, const CSTElement& node) {
-  FMT_ASSERT(node.isNode(), node,
-             "Expected node, Saw <" + std::string(symbolName(node.id())) + ">");
-  return count_leading_newlines(traits, node) + count_trailing_newlines(traits, node);
+// Determines if a given node would emit a leading comment if emitted.
+static bool has_leading_comment(const CSTElement& node, const token_traits_map_t& traits) {
+  return count_leading_newlines(traits, node) > 0;
 }
 
-// Counts all newlines allowd before and after a require header. Ignores the require body
-// - require a = b # comment -> 1
+// Determines if a given node would emit a trailing comment if emitted.
+static bool has_trailing_comment(const CSTElement& node, const token_traits_map_t& traits) {
+  return count_trailing_newlines(traits, node) > 0;
+}
+
+// Determines if the doc is "weakly flat". Weakly flat is a flat doc
+// with a single trailing comment allowed. No other newlines my be emitted.
+static bool is_weakly_flat(const wcl::doc& doc, const CSTElement& node,
+                           const token_traits_map_t& traits) {
+  return !doc->has_newline() ||
+         (doc->newline_count() == 1 && count_trailing_newlines(traits, node) == 1);
+}
+
+// Determizes of a doc is "flat" as a human would judge it. Human's don't
```

Comment:
"if" not "of"

Reply:
Also "Determizes" is spelled wrong. I noticed the "z" but I didn't trust my own spelling enough honestly.